### ISSUE TITLE
User Operations

### ISSUE THREAD

**Issue by rufer7** [dfensgmbh/biz.dfch.CS.Abiquo.Client#20]
Implement the following operations for `Users`

* [x] Read (Including `Role`)

**Comment by rufer7** [comment]
User objects of type `userswithroles` can only be requested when querying all users of an enterprise. When requesting a user by Id with Accept header `application/vnd.abiquo.userwithroles` the request fails with `406 Not Acceptable`. When requesting user by Id only Accept header `application/vnd.abiquo.user` is accepted by the API. The response of type `application/vnd.abiquo.user` contains the role of the user as link as shown below

```json
{
    "links": [
        {
            "title": "Abiquo",
            "rel": "enterprise",
            "type": "application/vnd.abiquo.enterprise+json",
            "href": "https://abiquo/api/admin/enterprises/1"
        },
        {
            "title": "CLOUD_ADMIN",
            "rel": "role",
            "type": "application/vnd.abiquo.role+json",
            "href": "https://abiquo/api/admin/roles/1"
        },
        {
            "title": "admin",
            "rel": "edit",
            "type": "application/vnd.abiquo.user+json",
            "href": "https://abiquo/api/admin/enterprises/1/users/1"
        },
        {
            "title": "virtual machines",
            "rel": "virtualmachines",
            "type": "application/vnd.abiquo.virtualmachines+json",
            "href": "https://abiquo/api/admin/enterprises/1/users/1/action/virtualmachines"
        },
        {
            "title": "pending tasks",
            "rel": "pendingtasks",
            "type": "application/vnd.abiquo.tasks+json",
            "href": "https://abiquo/api/admin/enterprises/1/users/1/action/pendingtasks"
        },
        {
            "title": "applications",
            "rel": "applications",
            "type": "application/vnd.abiquo.applications+json",
            "href": "https://abiquo/api/admin/enterprises/1/users/1/applications"
        }
    ],
    "id": 1,
    "name": "Arbitrary name",
    "nick": "Arbitrary nick",
    "locale": "en_US",
    "surname": "Arbitrary surname",
    "active": true,
    "email": "",
    "description": "Arbitrary description",
    "firstLogin": false,
    "locked": false,
    "authType": "Arbitrary AuthType"
}
```

According @dfch the role of the user will be contained in the JWT. If this will not be the case we could still parse the role name out of the link.

**Comment by rufer7** [comment]
resolved by #27
